src/d3a/models/market/future.py

```diff
@@ -0,0 +1,217 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from copy import deepcopy
+from logging import getLogger
+from typing import Dict, List, Union
+
+from d3a.d3a_core.exceptions import (BidNotFoundException, MarketReadOnlyException,
+                                     OfferNotFoundException)
+from d3a.events.event_structures import MarketEvent
+from d3a.models.market import lock_market_action
+from d3a.models.market.two_sided import TwoSidedMarket
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from d3a_interface.data_classes import Bid, Offer, Trade, BaseBidOffer
+from pendulum import DateTime, duration
+
+log = getLogger(__name__)
+
+
+class FutureMarketException(Exception):
+    """Exception specific to the Future markets."""
+
+
+class FutureMarkets(TwoSidedMarket):
```

Comment:
Can you explain why we name it in plural form?

Reply:
Scratch that, I understood now the reason and rational behind this implementation